gpflow/models/mixins.py

```diff
@@ -44,3 +44,11 @@ def closure():
             return training_loss(batch)
 
         return closure
+
+
+class MCMCTrainingLossMixin(InternalDataTrainingLossMixin):
```

Comment:
i don't know what MCMC is. clearly there's something about MCMC that means it has a different training loss. Should this also be conveyed in the names of the classes `InternalDataTrainingLossMixin` and the other one? Why, for example, don't they use the `log_posterior_density`? what's special about them?

Reply:
log_posterior_density is the right thing to compute for an MCMC model. maximum likelihood / map objective is the right thing to compute for the other models.

Reply:
what i wonder is if we can be a bit more be a bit more descriptive in the names of the other mixins